source/common/common/hash.h

```diff
@@ -62,6 +61,21 @@ class HashUtil {
     return XXH64(buf, sizeof(input), seed);
 #endif
   }
+  template <typename FloatingPoint,
+            std::enable_if_t<std::is_floating_point_v<FloatingPoint>, bool> = true>
+  static uint64_t xxHash64FloatingPoint(FloatingPoint input, uint64_t seed = 0) {
+    if (std::isnan(input)) {
+      return XXH64("NaN", 3, seed);
+    }
+    if (std::isinf(input)) {
+      return XXH64("Inf", 3, seed);
+    }
+    int exp;
+    auto frac = std::frexp(input, &exp);
+    seed = xxHash64Value(exp, seed);
+    uint64_t mantissa = frac * 9223372036854775808.0; // 2^63
```

Comment:
add link to where this formula comes from?

Reply:
oh i see; it's just that you get a float fraction and you are trying to turn it into a uint64 using the whole range. Cool. Maybe just write that as a comment?